### PR TITLE
Add an Ansible role to detect JBoss EAP 5

### LINKED ISSUE #848: Create Ansible role to determine the JBoss EAP 5 presence
## Specify type:
- Enhancement

## Description:
Create a new role (or add any additional tasks to the existing eap role) to capture EAP 5 presence.
Utilize standard non-invasive search:
- rpm
- process list
- init files (chckconfig, systemctl)
- documented installation paths

## Acceptance Criteria: 

- [ ] Verify that tasks specific to capturing JBoss EAP 5 
___

Associated with:  https://github.com/quipucords/quipucords/issues/847

### PR DESCRIPTION
**Pull request by noahl**
Tasks were chosen by installing EAP 5 on a test server and looking for
characteristics that could identify it.

Closes #848 .

### CODE REVIEW

File: roles/jboss_eap5/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  ignore_errors: yes
+  with_items: "{{ eap5_home_candidates }}"
+  when: 'jboss_eap'
+
+- name: EAP_HOME/jboss-as/bin/run.jar
+  raw: java -jar "{{ item }}/jboss-as/bin/run.jar --version"
+  register: eap5_home_run_jar_version
+  ignore_errors: yes
+  with_items: "{{ eap5_home_candidates }}"
+  when: 'jboss_eap'
```
Comment (chambridge) [inline roles/jboss_eap5/tasks/main.yml]: Should have:
`when: 'have_java and jboss_eap'`

**Comment by noahl** [comment]
Just a minute, I might need to update the role a bit.

**Comment by noahl** [comment]
@chambridge thanks! I'll post the output in a minute. I'm working on postprocessing and I might want to change this file a bit first.

**Comment by noahl** [comment]
@chambridge I just pushed an update with a slight change in the role. It's ready for review now.

I took the raw log entries from a scan and cleaned them up a bit. Here is that output:

Results for sonar-jbosseap-rhel-5-vc (has EAP 5):

```
$ locate JBossEULA.txt | sed -n -e \"s/\\(.*\\)\\/jboss-as\\/\\(.*\\)/\\1/gp\" | uniq
/opt/jboss/jbosseap-5.2.0/jboss-eap-5.2

$ ps -A -o comm -o args e --columns 10000 | egrep '^java.*(eap|jboss).*' | tr -s ' =:' '\\n' | sed -n -e \"s/\\(.*\\)\\/jboss-as\\/\\(.*\\)/\\1/gp\" | uniq
/opt/jboss/jbosseap-5.2.0/jboss-eap-5.2

$ cat /opt/jboss/jbosseap-5.2.0/jboss-eap-5.2/version.txt 2>/dev/null
JBoss Enterprise Application Platform - Version 5.2.0 GA

$ cat /opt/jboss/jbosseap-5.2.0/jboss-eap-5.2/readme.html 2>/dev/null
<html>
  <head>
    <title>JBoss Enterprise Application Platform 5.2</title>
  </head>
  <body>
  <p>
    All documentation for JBoss Enterprise Application Platform 5.2, 
    including Release Notes, is available from:<br/>
<a href=\"http://docs.redhat.com\">http://docs.redhat.com</a>
  </p>
  <p>
    Select <b>JBoss Enterprise Application Platform</b> &lt; <b>5</b> 
    from the sidebar for a full list of documentation available with 
    JBoss Enterprise Application Platform 5 and its patch releases.
  </p>

  </body>
</html>

$ ls -1 /opt/jboss/jbosseap-5.2.0/jboss-eap-5.2/jboss-as
bin
client
common
copyright.txt
docs
icons
jar-versions.xml
JBossEULA.txt
lgpl.html
lib
scripts
server
tools

$ java -jar /opt/jboss/jbosseap-5.2.0/jboss-eap-5.2/jboss-as/bin/run.jar --version
JBoss 5.2.0 (build: SVNTag=JBPAPP_5_2_0 date=201211232041)

Distributable under LGPL license.
See terms of license at gnu.org.
```

Results for sonar-jbosseap-rhel-7-vc, which has EAP 7 but not EAP 5:

```
$ locate JBossEULA.txt | sed -n -e \"s/\\(.*\\)\\/jboss-as\\/\\(.*\\)/\\1/gp\" | uniq
$ ps -A -o comm -o args e --columns 10000 | egrep '^java.*(eap|jboss).*' | tr -s ' =:' '\\n' | sed -n -e \"s/\\(.*\\)\\/jboss-as\\/\\(.*\\)/\\1/gp\" | uniq

(The other commands are not run on this host because there are no
candidate EAP_HOME directories.)
```

The raw log entries are attached if you really want to see them.
[eap5-results.txt](https://github.com/quipucords/quipucords/files/1764571/eap5-results.txt)

**Comment by noahl** [comment]
Two of the Travis jobs failed with an error running "git checkout". It looks like there's a Travis outage. I have restarted, and will see if they complete this time.

**Pull request merged**